pkg/octavia/lb_mgmt_network.go

```diff
@@ -512,6 +514,245 @@ func findRouter(client *gophercloud.ServiceClient, log *logr.Logger) (*routers.R
+	//
+	// Strip description out of search. While informative, we are not concerned with that field.
+	//
+	listOpts := *criteria
+	listOpts.Description = ""
+
+	allPages, err := rules.List(client, listOpts).AllPages()
+
+	if err != nil {
+		log.Error(err, "findSecurityGroupRule: Unable to find security group rule")
+		return nil, err
+	}
+	allRules, err := rules.ExtractRules(allPages)
+	if err != nil {
+		log.Error(err, "findSecurityGroupRule: error extracting security group rule")
+		return nil, err
+	}
+	if len(allRules) != 0 {
+		return &allRules[0], nil
+	}
+
+	return nil, nil
+}
+
+func strToRuleEtherType(v string) rules.RuleEtherType {
+	if v == "IPv6" {
+		return rules.EtherType6
+	}
+	return rules.EtherType4
+}
+
+func strToRuleProtocol(p string) rules.RuleProtocol {
+	switch p {
+	case "tcp":
+		return rules.ProtocolTCP
+	case "udp":
+		return rules.ProtocolUDP
+	}
+	return ""
+}
+
+func ensureSecurityGroupRules(client *gophercloud.ServiceClient, securityGroup *groups.SecGroup, rulesDefinitions []rules.ListOpts, log *logr.Logger) error {
+	for _, r := range rulesDefinitions {
+		r.TenantID = securityGroup.TenantID
+		r.SecGroupID = securityGroup.ID
+		r.Direction = "ingress"
+		rule, err := findSecurityGroupRule(client, &r, log)
+		// Don't break on error if not found, but create the rest.
+		if err != nil {
+			log.Error(err, fmt.Sprintf("ensureSecurityGroupRules: error searching for %s", r.Description))
+			continue
+		}
+		if rule != nil {
+			// There is a rule that matches this description, carry on.
+			continue
+		}
+		//
+		// Rule not found. Create a new one.
+		createOpts := rules.CreateOpts{
+			Description:  r.Description,
+			EtherType:    strToRuleEtherType(r.EtherType),
+			PortRangeMax: r.PortRangeMax,
+			PortRangeMin: r.PortRangeMin,
+			Protocol:     strToRuleProtocol(r.Protocol),
+			SecGroupID:   securityGroup.ID,
+			Direction:    rules.DirIngress,
+			ProjectID:    securityGroup.TenantID,
+		}
+		_, err = rules.Create(client, createOpts).Extract()
+		if err != nil {
+			log.Error(err, fmt.Sprintf("ensureSecurityGroupRules: error creating rule %s", r.Description))
+		}
+	}
+	return nil
+}
+
+func ensureMgmtRules(client *gophercloud.ServiceClient, securityGroup *groups.SecGroup, log *logr.Logger) error {
+	rulesDefinitions := []rules.ListOpts{
+		{
+			Description:  "ssh port IPv4 rule",
+			PortRangeMax: 22,
+			PortRangeMin: 22,
+			EtherType:    "IPv4",
+			Protocol:     "tcp",
+		},
+		{
+			Description:  "ssh port IPv6 rule",
+			PortRangeMax: 22,
+			PortRangeMin: 22,
+			EtherType:    "IPv6",
+			Protocol:     "tcp",
+		},
+		{
+			Description:  "amphora agent port IPv4 rule",
+			PortRangeMax: 9443,
+			PortRangeMin: 9443,
+			EtherType:    "IPv4",
+			Protocol:     "tcp",
+		},
+		{
+			Description:  "amphora agent port IPv6 rule",
+			PortRangeMax: 9443,
+			PortRangeMin: 9443,
+			EtherType:    "IPv6",
+			Protocol:     "tcp",
+		},
+		{
+			Description:  "log offloading tcp IPv4 rule",
```

Comment:
Ports 514 (UDP/TCP) should be opened in the HealthManager SG (log offloading packets come from the amp to the management ports, same as the heartbeat packets on port 5555)

Reply:
Ack thanks, fixing.